runtime/serde/serde-cbor/common/src/aws/smithy/kotlin/runtime/serde/cbor/encoding/Simple.kt

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.smithy.kotlin.runtime.serde.cbor.encoding
+
+import aws.smithy.kotlin.runtime.io.*
+import aws.smithy.kotlin.runtime.serde.DeserializationException
+import aws.smithy.kotlin.runtime.serde.cbor.encodeArgument
+import aws.smithy.kotlin.runtime.serde.cbor.encodeMajorMinor
+import aws.smithy.kotlin.runtime.serde.cbor.toULong
+
+/**
+ * Represents a CBOR string (major type 3) encoded as a UTF-8 byte array.
+ * @param value The [String] which this CBOR string represents.
+ */
+internal class String(val value: kotlin.String) : Value {
```

Comment:
**Question**: Since we're matching these to CBOR terms, should this be `TextString` (ref: [RFC 8949 § 3.1, Major type 3](https://www.rfc-editor.org/rfc/rfc8949.html#section-3.1-2.8))?